ci: compliance: add a PR description check step

fabiobaltieri opened the pull request:
Add an extra workflow step to block PRs with an empty description so that these gets flagged automatically and reviewers don't have to ask for it.

Add a message in the actual fail step as well so it's easier for the user to figure out what went wrong, even in the existing compliance check case.

--

Test run: https://github.com/zephyrproject-rtos/zephyr/actions/runs/9896115700/job/27337556338

Goes with zephyrproject-rtos/zephyr#75115

nashif commented:
so it adds DNM if body is empty? why not add this check to compliance, maybe it belongs better than this do_no_merge workflow.

fabiobaltieri commented:
> so it adds DNM if body is empty? why not add this check to compliance, maybe it belongs better than this do_no_merge workflow.

Yeah it's an option but the check compliance script itself does not currently do anything with the actual PR metadata, just the commit, and this is easy enough to check with just the action logic so I placed it here.

Maybe I can move the step in the compliance workflow, flag it as `continue-on-error` and check it as part of the `check-warns` step? Not sure how to do it but I can look it up.

fabiobaltieri commented:
Yeah let me experiment a bit and move this under the compliance workflow.

fabiobaltieri committed:
ci: compliance: add a PR description check step

Add an extra workflow step to block PRs with an empty description so that
these gets flagged automatically and reviewers don't have to ask for it.

Add a message in the actual fail step as well so it's easier for the
user to figure out what went wrong, even in the existing compliance
check case.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri commented on .github/workflows/compliance.yml:
```diff
@@ -1,6 +1,12 @@
+  pull_request:
+    types:
+    - edited
+    - opened
+    - reopened
+    - synchronize
```
these three should be the default, so I just added "edited"

fabiobaltieri commented:
Ok this is now in the compliance workflow, which makes a lot of sense. Added a couple error messages to help the user understand what's going on as well, that was an existing problem with the compliance run where the errors are in a different step, always found it a bit confusing, hopefully this takes care of that as well.

nashif commented:
wait a sec, does this mean we now will run all of compliance check workflow EVERY time someone edits the PR body, or title?

fabiobaltieri commented:
> wait a sec, does this mean we now will run all of compliance check workflow EVERY time someone edits the PR body, or title?

Well yeah of course, how else could it do it. I don't think it's a big deal, people edit descriptions way less often than force pushes and the compliance check is not that heavy, but I can always move it back to dnm or make a new workflow if you think that's a problem.

nashif commented:
> Well yeah of course, how else could it do it. I don't think it's a big deal, people edit descriptions way less often than force pushes and the compliance check is fairly lightweight, but I can always move it back to dnm or make a new workflow if you think that's a problem.

ok, lets see how things go, we can move to own workflow if needed.
